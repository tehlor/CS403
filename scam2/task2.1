(define (f a b) (+ a b))
(inspect ((curry f) 1))
