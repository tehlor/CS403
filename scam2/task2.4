(define (g a b c d) (- a b c d))
(inspect ((curry g) 28 19 9 4))
