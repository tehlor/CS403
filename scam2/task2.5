(define (g a b c d) (- a b c d))
(inspect ((curry g 39) 10 9 1))
