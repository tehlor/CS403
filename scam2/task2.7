(define (g a b c d) (- a b c d))
(inspect ((curry g 23 13) 9 3))
