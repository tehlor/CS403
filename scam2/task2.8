(define (g a b c d) (- a b c d))
(inspect (((curry g 25) 14) 9 4))
