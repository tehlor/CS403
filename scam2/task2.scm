(define (main)
    (setPort (open (getElement ScamArgs 1) 'read))
    (readArgs)
)

(define (readArgs)
    (define env this)
    (define (iter expr)
        (if (not (eof?))
            (begin (eval expr env) 
                (iter (readExpr))
            )
        )
    )
    (iter (readExpr))
)

(define (curry @)
    (if (is? (car @) 'CONS)
        (define x 
            (car @)
        )
        (define x @)
    )
    (define (continue) 
        (length (get 'parameters (car x)))
    )
    (if (< (length (cdr x)) (continue))
        (lambda (@) (curry (append x @)))
        (apply (car x) (cdr x))
    )
)