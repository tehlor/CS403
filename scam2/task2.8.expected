(((curry g 25) 14) 9 4) is -2
